instrumentation/build.gradle.kts

```diff
@@ -29,6 +28,20 @@ android {
         }
     }
 
+    libraryVariants.all {
+        outputs
+            .map { it as com.android.build.gradle.internal.api.BaseVariantOutputImpl }
```

Comment:
While this works, I'm a bit concerned that it's using an internal type, which itself uses a deprecated API. So I think we should instead use the new `androidComponents` API, which has a few handy utilities for build-manipulation use cases.

For this case, I think we can even get an `aar` file without any suffixes (debug or release) to make it easier to tell what's the variant we plan to release. We could do it like so:

```kotlin
androidComponents {
    onVariants {
        if (it.buildType == "release") { // The one we choose to release
            project.tasks.register("createReleaseBuild", Copy::class) {
                from(it.artifacts.get(com.android.build.api.artifact.SingleArtifact.AAR))
                into(project.layout.buildDirectory.dir("outputs/aar"))

                rename(".+", "opentelemetry-android.aar")
            }
        }
    }
}
```

That way, whenever we need to create a release build, we can run the task `createReleaseBuild` (or any other name we come up with) which will generate the chosen variant and set the proper file name to it. We can also define any other folder to place it, it doesn't necessarily have to be "build/outputs/aar".

Reply:
Thanks, my approach was definitely a hack job and it feels gross. I definitely struggle with this gradle magic incantation stuff. I couldn't find any reasonable documentation whatsoever about the idiomatic way to rename/customize android library aars using the gradle plugin. 

I tried your approach and it seems to work just fine and I think we should use it. I wish it wasn't a new task name and I wish it was just part of the normal `aseemble` process.

Reply:
I agree, Gradle can be very unintuitive sometimes 😅 

> I wish it was just part of the normal aseemble process

Well, I just tried a couple of things and it seems like this could make it possible:

```kotlin
androidComponents {
// ...
}

project.afterEvaluate {
    tasks.named("assembleRelease") {
        finalizedBy("createReleaseBuild")
    }
}
```

Don't know if it's too "hacky" though... So anyway, I'll leave it here in case you want to add it 😬

Reply:
> I agree, Gradle can be very unintuitive sometimes 😅

You are being very generous. I'd usually say something much much worse. 🙃 

In any case, this is great. Doesn't feel hacky to me. Thank you!